subprojects/core/src/integTest/groovy/org/gradle/api/internal/cache/CacheConfigurationsIntegrationTest.groovy

```diff
@@ -77,20 +120,69 @@ class CacheConfigurationsIntegrationTest extends AbstractIntegrationSpec {
                 settings.caches {
                     markingStrategy = MarkingStrategy.NONE
                     cleanup = Cleanup.DISABLED
-                    releasedWrappers.removeUnusedEntriesOlderThan = ${releasedDistTimestamp}
-                    snapshotWrappers.removeUnusedEntriesOlderThan = ${snapshotDistTimestamp}
-                    downloadedResources.removeUnusedEntriesOlderThan = ${downloadedResourcesTimestamp}
-                    createdResources.removeUnusedEntriesOlderThan = ${createdResourcesTimestamp}
+                    releasedWrappers.entryRetentionAge = ${releasedDistTimestamp}L
+                    snapshotWrappers.entryRetentionAge = ${snapshotDistTimestamp}L
+                    downloadedResources.entryRetentionAge = ${downloadedResourcesTimestamp}L
+                    createdResources.entryRetentionAge = ${createdResourcesTimestamp}L
+                    buildCache.entryRetentionAge = ${buildCacheResourcesTimestamp}L
                 }
             }
         """
         settingsFile << """
             caches {
                 assert markingStrategy.get() == MarkingStrategy.NONE
-                assert releasedWrappers.removeUnusedEntriesOlderThan.get() == ${releasedDistTimestamp}
-                assert snapshotWrappers.removeUnusedEntriesOlderThan.get() == ${snapshotDistTimestamp}
-                assert downloadedResources.removeUnusedEntriesOlderThan.get() == ${downloadedResourcesTimestamp}
-                assert createdResources.removeUnusedEntriesOlderThan.get() == ${createdResourcesTimestamp}
+                assert releasedWrappers.entryRetentionAge.get() == ${releasedDistTimestamp}
+                assert snapshotWrappers.entryRetentionAge.get() == ${snapshotDistTimestamp}
+                assert downloadedResources.entryRetentionAge.get() == ${downloadedResourcesTimestamp}
+                assert createdResources.entryRetentionAge.get() == ${createdResourcesTimestamp}
+                assert buildCache.entryRetentionAge.get() == ${buildCacheResourcesTimestamp}
+            }
+        """
+
+        expect:
+        succeeds("help")
+    }
+
+    def "can configure caches to a custom entry retention timestamp"() {
```

Comment:
❔ Did you add a test that shows that the retention timestamp takes precedence over the retention age? I didn't see one at first glance.

Reply:
I intended to, but I guess that slipped. Do you think a unit test would be sufficient?
(We could verify that the returned `entryRetentionTimestampSupplier` uses the timestamp in preference to the age.

Note that setting the `entryRetentionTimestamp` is still an internal API. We could consider making it public, in which case test coverage will be more important.

Reply:
Similar to my previous comment, I think it would be fine to remove this unless we're not testing this functionality elsewhere (i.e. a unit test would be fine IMO).